chalice/app.py

```diff
@@ -379,6 +429,20 @@ def _register_view(view_func):
             return view_func
         return _register_view
 
+    def _validate_route_content_types(self, name, content_types):
+        # A routes content_types be homogeneous in their binary support.
+        binary, non_binary = [], []
+        for content_type in content_types:
+            if content_type in self.api.binary_types:
+                binary.append(content_type)
+            else:
+                non_binary.append(content_type)
+        if binary and non_binary:
+            raise ValueError(
+                'In view function "%s", the content_types %s support binary '
+                'and %s do not. All content_types must be consistent in their '
```

Comment:
Not technically, but it changes the interface and I couldn't think of any good use to have that rather than just writing two separate routes to handle your data accordingly. And it could be a source of bugs if you are not careful, since if you do not read the documentation carefully and you mix the types you may get rare crashes which may not be immediately clear.

I believe if you send a bytes type to lambda and it isn't base64 encoded properly then you get a "circular reference" error during json encoding which isn't very clear. As a user of chalice I think I would prefer to simply be called out on this ahead of time and be forced to separate my logic into two routes.

Maybe there is a common use case I'm not thinking of?